lib/goth/server.ex

```diff
@@ -49,8 +32,25 @@ defmodule Goth.Server do
         Goth.HTTPClient.init({module, opts})
       end)
 
+    {async, opts} = Keyword.pop(opts, :async)
+
     state = struct!(__MODULE__, opts)
 
+    if async do
+      {:ok, state, {:continue, :async_fetch}}
```

Comment:
let's use the name _prefetch_

```suggestion
      {:ok, state, {:continue, :async_prefetch}}
```